core/org.wso2.carbon.server/src/main/java/org/wso2/carbon/server/Main.java

```diff
@@ -69,14 +69,12 @@ public static void main(String[] args) {
 
 		}
         writePID(System.getProperty(LauncherConstants.CARBON_HOME));
-        //setting default WSO2CarbonProfile as the running Profile if no other Profile is given as an argument
-        if (System.getProperty(LauncherConstants.PROFILE_ID) == null){
-            System.setProperty(LauncherConstants.PROFILE_ID, LauncherConstants.DEFAULT_CARBON_PROFILE);
-        }
         processCmdLineArgs(args);
 
         // set WSO2CarbonProfile as worker if workerNode=true present
-        if ((System.getProperty(LauncherConstants.WORKER_NODE) != null) && (System.getProperty(LauncherConstants.WORKER_NODE).equals("true"))) {
+        if ((System.getProperty(LauncherConstants.WORKER_NODE) != null) && 
+            (System.getProperty(LauncherConstants.WORKER_NODE).equals("true")) &&
```

Comment:
Use; "true".equals(System.getProperty(LauncherConstants.WORKER_NODE)) that way you don't need a null check.